### PR TITLE
Create basic instrument generation example

### PR DESCRIPTION
**Pull request by ryaanahmed**
This PR adds an example of using the sliders and file upload from #21 to plug into a synthesis module and sonify the data by using a hardcoded 'ratios' column to generate the pitch set for an interactive instrument.

SUPER basic, not ready for any kind of primetime, but here as an example of how to put this stuff together using the existing systems.

https://user-images.githubusercontent.com/691117/133824686-9bd2a750-cce6-4448-8c1e-40579361d6c0.mp4

### CODE REVIEW

**Comment by ryaanahmed** [comment]
Also some minor fixes:
- Add slashes to the end of all of our routes in `urls.py`: this allows Django to automatically redirect non-slash routes to slash. It also means we have to be strict about terminating API routes with slashes, as it can't redirect when there's POST data. I think worth being strict about this.
- I realized here that we were missing the babel `transform-runtime` plugin, which I think we might need for `useEffect` + async stuff if we end up using AudioContext. Better to add now as folks are still doing setup than later. @anastasia if folks run into build issues, they'll need to run `npm ci` again.
- Some minor tweaking to linter settings.

File: backend/app/views.py [hunk 1]
```diff
@@ -186,21 +188,24 @@ def image_to_music(request):
...
-@api_view(['POST'])
-def change_settings(request):
-    """
-    Handling settings changes from frontend
-    """
-    print("type:", request.data.get('type'),
-          "id:", request.data.get('id'),
-          "value:", int(request.data.get('value')))
-    return Response(request.data)
+def generate_instrument(request):
```
Comment (ryaanahmed) [inline backend/app/views.py:191]: The bulk of this probably shouldn't be here in a view function, but doing this quick and dirty for now. Will refactor next week.

File: frontend/components/Home.js [hunk 2]
```diff
@@ -1,22 +1,81 @@
...
-import VerticalSlider from "./VerticalSlider";
+import Slider from "./Slider";
 
-const Home = () => {
-    const verticalSlidersNum = 9;
-    const verticalSlidersList = [];
-    for (let i = 0; i < verticalSlidersNum; i++) {
-        verticalSlidersList.push(
+
+const Instrument = ({samples}) => {
```
Comment (ryaanahmed) [inline frontend/components/Home.js:7]: As above - this should move.

File: frontend/components/Home.js [hunk 3]
```diff
@@ -1,22 +1,81 @@
...
         <>
             <h2>Welcome to our Sonification toolkit!</h2>
-
-            <UploadFileInput id={0}/>
-            <ul className="list-inline">{verticalSlidersList}</ul>
+            <UploadFileInput
+                id={0}
+                uploadSuccessfulCallback={setInstrumentSamples}
+                apiEndpoint={'api/generate_instrument/'}
+            />
```
Comment (anastasia) [inline frontend/components/Home.js:75]: nice

File: backend/app/views.py [hunk 4]
```diff
@@ -186,21 +188,24 @@ def image_to_music(request):
...
+    """
+    Takes a 1-D CSV with the 'ratio' header and constructs samples based on those
+    ratios.
+    """
+    temp_file = request.FILES.get('value')
+    csv_data = encoders.parse_csv_upload(temp_file)
+    ratios = [float(row['ratio']) for row in csv_data]
```
Comment (anastasia) [inline backend/app/views.py:198]: we should have a try catch with error reporting to the frontend here.

**Review by anastasia — APPROVED** [review]

**Pull request merged**